### ISSUE TITLE
i run “SkipCTNN.py”， got the accuracy just “0.32” on training set

### ISSUE THREAD

**Issue by xupinjie** [SailColubrid/tnn#1]
i run “SkipCTNN.py”， got the accuracy just “0.32” on training set

**Comment by xupinjie** [comment]
i run “CTNN.py”， got the accuracy just “0.822” on training set，got the accuracy just “0.803” on test set.
could you please give me some advise？
Thakns

**Comment by zifanw** [comment]
Sorry for the confusion. So the SkipCTNN is still under test so you may choose to ignore it. We basically use CTNN.py script to generate our result. 

I update some parts of the code since I realize I have commented the training part of SVM previously. 

But if you don't want to pull and remain as the previous version, you can:
1) So the first thing you can try is to retrain the CTNN by uncommenting line 253 and 254:

`#net = train(net, MNIST_loader)`
`#torch.save(net.state_dict(), "./MNISTcheckpoint.pt")`
It may take some time depending on your machine. 

2) Uncomment the part to train SVM.
`clf.fit(train_outputs, train_y)`

I remember this 0.822 appears once in my experiments. So in the training, to speed up the process, I use less data to train. You may try use more data by changing the line 159 and 165 in my updated version:
`if np.random.random() > 0.2:`

So 0.2 means that only 80% data are used for training. You can play with this number to try

**Comment by xupinjie** [comment]
@SailColubrid  Thanks for your help.
I run your code summit 3 days ago. But I still  get the result 0.84.
I have some more problem.
In "CTNN.py"
line 231: I could not find the OnCenter and OffCenter functions in your branch "miladmozafari/SpykeTorch-@ c13b394"
line 54: Why the input channel of the first spike-convolution layer is 4? The 4 input channels may cause error.

I change the input channels of first con-layer to 2, using DoGKernel without OnCenter and OffCenter, getting the results 0.84 on training set, and 0.81 on test set. 
If comment `if np.random.random() > 0.2:`, the result is 0.83 on training set, 0.81 on test set.

Could you please give me some advice about above issues?

**Comment by zifanw** [comment]
What do you mean by "submitting 3 days ago?" I didn't see any pull request. 

Sorry for the ON-Center / OFF-Center filters. I remembered. So they are not supported by the SpykeTorch group and are programmed by us. Yeah this is a serious problem since I use the submodule instead of uploading our revised Spyketorch module. I will update this issue layer. So this is also the answer to "why the input channel is 4" Since you have four filters(two DoG and two Centers) and the input image has only one channel. So you will have 1 x 4 = 4 channels first. But if you only use two DoG filters, you just need 2 channels. In my experiments, adding ON-Center/OFF-Center filters make the accuracy higher!

**Comment by xupinjie** [comment]
@SailColubrid Thank  you very much. Would you mind update the `Spyketorch module`? This is important for my experiments. Thanks！

**Comment by zifanw** [comment]
Pls go and check out the latest update.
